android/src/main/java/com/imagepicker/ImagePickerModule.java

```diff
@@ -547,10 +553,16 @@ private boolean permissionsCheck(@NonNull final Activity activity,
     final int cameraPermission = ActivityCompat
             .checkSelfPermission(activity, Manifest.permission.CAMERA);
 
-    final boolean permissionsGrated = writePermission == PackageManager.PERMISSION_GRANTED &&
-            cameraPermission == PackageManager.PERMISSION_GRANTED;
+    boolean permissionsGranted = false;
 
-    if (!permissionsGrated)
+    switch (requestCode) {
+      case REQUEST_PERMISSIONS_FOR_LIBRARY:
+        permissionsGranted = writePermission == PackageManager.PERMISSION_GRANTED;
+      case REQUEST_PERMISSIONS_FOR_CAMERA:
```

Comment:
You're missing a break statement here, which causes the wrong 'permission' to be checked when selecting image from library

Reply:
@Johan-dutoit fixed!